plutus-th/src/Language/Plutus/TH.hs

```diff
@@ -0,0 +1,19 @@
+{-# LANGUAGE TemplateHaskell #-}
+module Language.Plutus.TH (plutus, plutusT, PlcCode, getSerializedCode, getAst) where
+
+import           Language.Plutus.CoreToPLC.Plugin
+
+import           Language.Haskell.TH
+
+-- | Covert a quoted Haskell expression into a corresponding Plutus Core program. Produces an expression of type
+-- 'PlcCode'.
+plutus :: Q Exp -> Q Exp
+-- We would like to do `addCorePlugin "Language.Plutus.CoreToPLC.Plugin"``, but this needs template-haskell-2.13.0.0,
+-- which is in the next LTS snapshot.
+plutus e = [| plc $(e) |]
+
+-- | Covert a (typed) quoted Haskell expression into a corresponding Plutus Core program.
+plutusT :: Q (TExp a) -> Q (TExp PlcCode)
+-- We would like to do `addCorePlugin "Language.Plutus.CoreToPLC.Plugin"``, but this needs template-haskell-2.13.0.0,
+-- which is in the next LTS snapshot.
+plutusT e = [||plc $$(e)||]
```

Comment:
Also, we need to be careful with "new" features as we may have to initially deploy this via GHCJS, which AFAIK is still stuck at GHC 7.10.3.

Reply:
nixpkgs supports GHCJS 8.4 and 8.2: https://github.com/NixOS/nixpkgs/pull/40392

Reply:
Is this referring to typed splices (I think they're quite old), or `addCorePlugin`?

Reply:
@j-mueller Thanks. I was going off the GitHub page, which lists 7.10.3 as the default, but closer inspection reveals, there is a 8.0 branch.